### PR TITLE
warp cursor to center of window for `cursor_follows_focus`

### PR DESCRIPTION
**Pull request by m4rch3n1ng**
i'm not sure if you would want to put this behind a config, but i don't know if the cursor warping to the top left of a window is ever the better option over warping to the center, but i can config it if desirable.

i removed the `if is_some()` and then `unwrap()` thing, because `Rectangle` is copy and not borrowed anyway, but i don't exactly know why this was here so i can just revert it if necessary.

### CODE REVIEW

**Comment by git-f0x** [comment]
This is planned as the default behavior, so probably no config option necessary. (pop-os/cosmic-settings#567) :)

File: src/shell/focus/mod.rs [hunk 1]
```diff
@@ -231,29 +231,29 @@ fn update_focus_state(
...
                 let mut shell = state.common.shell.write().unwrap();
-                // get the top left corner of the target element
+                // get geometry of the target element
                 let geometry = shell.focused_geometry(target.unwrap());
-                //to avoid the nested mutable borrow of state
-                if geometry.is_some() {
-                    let top_left = geometry.unwrap().loc.to_f64();
+                if let Some(geometry) = geometry {
+                    // get the center of the target element
+                    let new_pos = (geometry.loc + geometry.size.downscale(2).to_point()).to_f64();
```
Comment (Drakulix) [inline src/shell/focus/mod.rs:238]: This is quite the nitpick, but `downscale(2).to_point()` doesn't really communicate the intent of this operation at all. Please do a more obvious devision by 2 or add a comment.

**Review by Drakulix — APPROVED** [review]
Thanks!

File: src/shell/focus/mod.rs (hunk 1 above)
Comment (m4rch3n1ng) [reply src/shell/focus/mod.rs]: done, i think

**Review by Drakulix — APPROVED** [review]
Thx!

**Pull request merged**